+bids/layout.m

```diff
@@ -353,17 +354,21 @@ function tolerant_message(tolerant, msg)
                             subject.perf(j).m0type = 'separate_scan';
                             subject.perf(j).m0explanation = 'M0 was obtained as a separate scan';
                             
-                            % M0scan.nii filename
-                            m0_filename = [subject.perf(j).filename(1:end-10) 'm0scan' subject.perf(j).ext]; % assuming the (.nii|.nii.gz) extension choice is the same throughout
-                            if ~exist(fullfile(pth, m0_filename), 'file')
+                            % Check if M0scan.nii exists
+                            m0_filebase = bids.internal.file_utils(bids.internal.file_utils(subject.perf(j).filename, 'basename'), 'basename'); % remove the extension
+                            m0_filebase = m0_filebase(1:end-4); % to remove the _asl suffix
+                            m0_filename = bids.internal.file_utils('List', pth, sprintf('^%s.*_m0scan\\.nii(\\.gz)?$', m0_filebase)); % find the file
+                            m0_basename = bids.internal.file_utils(bids.internal.file_utils(m0_filename, 'basename'), 'basename'); % remove the extension
+                            
+                            if isempty(m0_filename)
                                 warning(['Missing: ' m0_filename]);
                             else
                                 % subject.perf(j).m0_filename = m0_filename;
                                 % -> this is included in the same structure for the m0scan.nii
                             end
                             
                             % M0 sidecar filename
-                            m0_json_sidecar_filename = [subject.perf(j).filename(1:end-10) 'm0scan.json'];
+                            m0_json_sidecar_filename = [m0_basename '.json'];
```

Comment:
@HenkMutsaerts 
there is no bids-example that covers the usecas "M0Type":"Separate", is there?
this would help with the tests: https://github.com/bids-standard/bids-matlab/blob/dev/tests/test_bids_query_asl.m